test/embedded/availability-code-removal.swift

```diff
@@ -0,0 +1,9 @@
+// RUN: %target-swift-frontend -emit-ir %s -parse-stdlib -enable-experimental-feature Embedded
+
+public protocol Player {}
+struct Concrete: Player {}
+
+@_unavailableInEmbedded
+public func test() -> any Player {
+  Concrete() // no error because we're in unavailable-in-embedded context
```

Comment:
> What is this test testing? Is it that the fact that test() uses existentials but that use is undiagnosed because the context is @_unavailableInEmbedded?

Exactly, and the breakdown is that it checks that (1) @_unavailableInEmbedded is taken into account and makes the declaration unavailable, (2) unavailable function body removal is on in embedded mode, (3) hence this particular function (that would be invalid in embedded Swift due to the usage of an existential) is allowed and not reported by the existential checker.

Let me comment that out explicitly in the test. And add the CHECK.